airflow/models/dagbag.py

```diff
@@ -248,7 +247,11 @@ def process_file(self, filepath, only_if_updated=True, safe_mode=True):
 
             with timeout(self.DAGBAG_IMPORT_TIMEOUT):
                 try:
-                    m = imp.load_source(mod_name, filepath)
+                    loader = importlib.machinery.SourceFileLoader(mod_name, filepath)
+                    spec = importlib.util.spec_from_loader(mod_name, loader)
+                    m = importlib.util.module_from_spec(spec)
+                    sys.modules[spec.name] = m
```

Comment:
We don't need this, do we?

Reply:
The following should also work:

```python
spec = importlib.util.spec_from_file_location(module_name, module_path)
module = importlib.util.module_from_spec(spec)
spec.loader.exec_module(module
```

Reply:
Thanks for you comment.
What you suggested is actually my first approach.
https://github.com/apache/airflow/pull/7099/commits/7fd424457a10e161e819bf376b601716e481f194

But it doesn't pass tests.

Reply:
Ok but you should not need "sys.modules[spec.name] = m"

Reply:
It's the second approach.
https://github.com/apache/airflow/pull/7099/commits/68c539fdcd74b900bcc60f441ac5ceb3b4f1b63d
That approach successfully finish some test-cases that the first one fails.
But still fails a test-case.
`test_backfill_job.py::TestBackfillJob::test_backfill_examples_2_example_skip_dag`

Reply:
We shouldn't be updating the sys.modules directly.

I will fallback to what other committers think @potiuk @ashb :

Reply:
FYI:
The way manipulating `sys.modules` directly is introduced in [the Python official document](https://docs.python.org/3/library/importlib.html#importing-a-source-file-directly).

Reply:
You are right, thanks for pointing out that link.